### PR TITLE
feat(marxan-runner): use output directory provided via input.dat

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Comment by kgajowy** [comment]
Most likely it is about time to consider refactor of Workspace to contain all fetches/creations.

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/workspace.service.ts [hunk 1]
```diff
@@ -21,8 +21,12 @@ export class WorkspaceService implements WorkspaceBuilder {
...
 
     console.log(`doing things in..`, workingDirectory);
-    return new Workspace(workingDirectory, marxanBinaryPath, async () =>
-      this.workingDirectoryService.cleanup(workingDirectory),
+    return new Workspace(
+      workingDirectory,
+      marxanBinaryPath,
+      async () => this.workingDirectoryService.cleanup(workingDirectory),
+      async () =>
+        this.workingDirectoryService.createOutputDirectory(workingDirectory),
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/workspace.service.ts:29]: why don't just pass an object here?
also, why isn't port abstract?

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts [hunk 2]
```diff
@@ -22,7 +23,9 @@ export class SharedStorage implements TemporaryDirectory {
   }
 
   async cleanup(directory: string): Promise<void> {
-    // TODO check if starts with tempDirectory
+    if (this.#hasPoisonNullByte(directory)) {
+      throw new Error(`Hacking is not allowed.`);
+    }
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts:28]: shouldn't we validate it on input, before cleaning up?

**Review by Dyostiq — APPROVED** [review]

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts (hunk 2 above)
Comment (kgajowy) [reply api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts:28]: @Dyostiq would you mind to elaborate? It is stateless, so you can pass any directory and it validates it on every use 🤔

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts [hunk 3]
```diff
@@ -32,10 +35,20 @@ export class SharedStorage implements TemporaryDirectory {
...
+    ) as WorkingDirectory;
+
     await promises.mkdir(resolve(this.#tempDirectory, directory));
-    // TODO replace output with the name from params
-    await promises.mkdir(resolve(this.#tempDirectory, directory, 'output'));
+
     return fullPath as WorkingDirectory;
   }
+
+  #hasPoisonNullByte = (path: string): boolean => path.indexOf('\0') !== -1;
```
[the remark this answers is not in the document]
Comment (kgajowy) [reply api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/workspace/shared-storage.ts:44]: @Dyostiq https://github.com/Vizzuality/marxan-cloud/pull/331/commits/49fc1b7c38013cbe6d9f776bf51bd64e7e3bc65e

**Pull request merged**